core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/elasticsearch/SnapshotsRule.scala

```diff
@@ -63,7 +63,7 @@ class SnapshotsRule(val settings: Settings)
       ) match {
         case NotAltered =>
           Fulfilled(blockContext)
-        case Altered(filteredSnapshots) if filteredSnapshots.nonEmpty && blockContext.requestContext.isReadOnlyRequest =>
+        case Altered(filteredSnapshots: Set[SnapshotName]) if filteredSnapshots.nonEmpty && blockContext.requestContext.isReadOnlyRequest =>
```

Comment:
`sealed trait AlterResult[+T]` was covariant. I should have changed that (and I did in many other places where covariance was used in the sealed trait hierarchy).

Type checks in pattern matching in Scala 3 are more strict. Scala 3 compiler is not sure of the type here - because of the redundant covariance. The `sealed trait AlterResult` generally should not be covariant. It was covariant only because we use `case object` typed with `Nothing`. Scala 2 compiler allowed that, but Scala 3 does not. We can have final case class instead and no variance, which will make type checks work.